spec/lib/jets/dotenv_spec.rb

```diff
@@ -0,0 +1,63 @@
+describe Jets::Dotenv do
+  describe "#load!" do
+    it "replaces ${ssm:<relative-path>} with SSM parameters prefixed with /<app-name>/<jets-env>/" do
+      expect(::Dotenv).to receive(:load).and_return(
+        "AUTENTICATED_URL" => "https://${ssm:username}:${ssm:password}@host.com",
+      )
+
+      username_response_double = double(
+        Aws::SSM::Types::GetParameterResult,
+        parameter: Aws::SSM::Types::Parameter.new(value: "my-user"),
+      )
+      expect_any_instance_of(Aws::SSM::Client).to receive(:get_parameter)
+        .with(name: "/demo/test/username", with_decryption: true)
+        .and_return(username_response_double)
+
+      password_response_double = double(
+        Aws::SSM::Types::GetParameterResult,
+        parameter: Aws::SSM::Types::Parameter.new(value: "my-password"),
+      )
+      expect_any_instance_of(Aws::SSM::Client).to receive(:get_parameter)
+        .with(name: "/demo/test/password", with_decryption: true)
+        .and_return(password_response_double)
+
+      env = Jets::Dotenv.new.load!
+
+      expect(env.fetch("AUTENTICATED_URL")).to eq "https://my-user:my-password@host.com"
+      expect(ENV.fetch("AUTENTICATED_URL")).to eq "https://my-user:my-password@host.com"
```

Comment:
It's important to confirm that both the returned `env` hash as well as global `ENV` has been mutated  since both are used in the Jets codebase.

That said it's definitely not good practice to mutate global ENV in specs since it can end up being a source of heisen-specs. Let me know if you have any preferences for reverting ENV to its original state after running the spec.

Reply:
Yes, can see how folks think it is not a good practice to change the ENV. Think this is fine for test also though.